wgpu-core/src/validation.rs

```diff
@@ -1290,3 +1329,23 @@ pub fn validate_color_attachment_bytes_per_sample(
 
     Ok(())
 }
+
+/// An entry point name finalized with [`Interface::finalize_entry_point_name`].
+#[derive(Debug)]
+pub struct FinalizedEntryPointName {
+    inner: String,
+}
+
+impl Deref for FinalizedEntryPointName {
+    type Target = str;
+
+    fn deref(&self) -> &Self::Target {
+        self.inner.as_ref()
+    }
+}
+
+impl FinalizedEntryPointName {
+    pub(crate) fn new(inner: String) -> Self {
+        Self { inner }
+    }
+}
```

Comment:
question: I'm not sure that this newtype is worth it. I'm considering getting rid of it, and just using `String`s after finalizing the entry point name.

Reply:
I would prefer to not have the newtype. I think that, as you said, converting from the option to the string is enough prevent us from forgetting that something has to happen.

Reply:
Removed with 20fa2a84f.